### ISSUE TITLE
NILMTK on Raspberry Pi: it is possible to install NILMTK on an architecture like Raspberry Pi 3b

### ISSUE THREAD

**Issue by drekisen** [nilmtk/nilmtk#918]
Hello greetings.

I've been trying to install NILMTK on a Raspberry Pi 3B for a while.

But at the time of the installation of NILMTK = 0.4.2 it does not allow me to install, I thought about installing packages individually, but still I can not install, so I have started to wonder if it is possible that it can be perform the installation, or it may not be possible on this type of card, such as the raspberry.
![instalacion de nilmtk en raspberry fallida](https://user-images.githubusercontent.com/72052079/110390177-34a35300-802b-11eb-9662-c49fd3ce8efd.JPG)

**Comment by drekisen** [comment]
Due to the architecture I have had to perform the installation of a miniconda instead of a conda and with this a possible decrease in the possibilities for the installation of NILMTK
![instalacion de nilmtk en raspberry fallida 2](https://user-images.githubusercontent.com/72052079/110390865-3de0ef80-802c-11eb-8336-5762dbffa469.JPG)
![instalacion de nilmtk en raspberry fallida 3](https://user-images.githubusercontent.com/72052079/110390867-3e798600-802c-11eb-8ba8-f329c2eeb654.JPG)

**Comment by PMeira** [comment]
Please see the comments from https://github.com/nilmtk/nilmtk/issues/895

scikit-learn and pandas are strict requirements. If you don't have those and can't build them yourself, I'm not sure it's worth trying to solve further issues.

For aarch64, I built packages for hmmlearn and I imagine you could built it from source for armv7l without issues. Psycopg2 is not required anymore since the Pecan Street Dataport converter is broken anyway.

**Comment by drekisen** [comment]
Greetings.

Thank you for the indications you provide.

Now I'm going to try something similar for the Raspberry Pi.

I already read all the information that was posted in the comments, although I'm just going to start experimenting

**Comment by PMeira** [comment]
@drekisen Hi, were you able to achieve any progress for RPi3?

I should be able to test some more both aarch64 and armv7 later this week (waiting for an SSD to arrive).

**Comment by drekisen** [comment]
Hi how are things.

According to 895 I followed the first installation of the packages and they all installed without problems.

First install the packages and attach them to a yml with which an environment was created and activated for the installation of nilmtk, then when I called "import nilmtk" within python or ipython it appeared that it could not find nilm_metadata.

I proceeded to download the nilm_metada folder, but I don't know where I should put that folder.


![carpeta](https://user-images.githubusercontent.com/72052079/114459887-b6e0e300-9ba6-11eb-8b3a-de3a1f36de51.JPG)

**Comment by drekisen** [comment]
So with python 3.6.6, which is the one that could be installed on the Raspberry, I decided to install the nilm_metadata through the following code, already with the environment activated and also nilmtk-0.4.0.dev1

![image](https://user-images.githubusercontent.com/72052079/114460217-1fc85b00-9ba7-11eb-8810-4b1909f11029.png)

**Comment by drekisen** [comment]
Through "conda list" it appears installed, both nilm_metada and nilmtk

![image](https://user-images.githubusercontent.com/72052079/114461696-abda8280-9ba7-11eb-977e-2f1d6ec4001e.png)

**Comment by drekisen** [comment]
But when placing "import nilmtk" in python it sends me a series of errors.
![image](https://user-images.githubusercontent.com/72052079/114461866-e512f280-9ba7-11eb-8322-2da7ee2b5097.png)

![image](https://user-images.githubusercontent.com/72052079/114461913-f22fe180-9ba7-11eb-9e9f-40881d7fe967.png)

I have not solved those problems anymore, I think my problem lies in nilm_metada, but I'm not sure about it

**Comment by PMeira** [comment]
Ah, about nilmtk_metadata, if you downloaded the very old file from the GitHub releases, that's completely broken nowadays.

You can either use the conda-package (preferred method)
```
conda install -c nilmtk nilmtk_metadata=0.2.4
``` 

or install directly from the actual latest release via pip:

```
pip install https://github.com/nilmtk/nilm_metadata/archive/refs/tags/0.2.4.zip
```

**Comment by PMeira** [comment]
Oops, I meant:

```
conda install -c nilmtk nilm_metadata=0.2.4
```

**Comment by drekisen** [comment]
Exactly, I did it as you indicated and apparently it has been installed successfully
![image](https://user-images.githubusercontent.com/72052079/114470025-c9154e00-9bb3-11eb-83d2-e4e5555ae400.png)

**Comment by drekisen** [comment]
But now when I do "import nilmtk"
It marks me that matplotlib is not installed, but when I give it "conda list" if it is found and when I install it again it tells me that it is already installed.

![image](https://user-images.githubusercontent.com/72052079/114470200-214c5000-9bb4-11eb-8edb-0c3d1a772f7c.png)

![image](https://user-images.githubusercontent.com/72052079/114470271-39bc6a80-9bb4-11eb-8d5c-1c4f245392ce.png)


![image](https://user-images.githubusercontent.com/72052079/114470321-4ccf3a80-9bb4-11eb-861b-e597ce1d15ae.png)

**Comment by drekisen** [comment]
Thanks for the help, I really appreciate it

**Comment by PMeira** [comment]
The kind of errors you described looks a lot like when two Python installations/environments get mixed somehow.

Did you try using matplotlib directly to see if it's working? That could give you more clues.

**Comment by PMeira** [comment]
From what I gather, that conda channel (rpi) is retired. The Pi3 only has 1 GB of RAM, right? That might be a limitation, but one option would be to try to use a 64-bit Linux. With that, I'm pretty sure most packages would work out of the box with conda-forge (I recommend minimamba).

I'm still going to try a 32-bit ARM installation to be sure, probably this weekend.

**Comment by Raphael164** [comment]
How can i accomplish an working installation on a raspberry Pi 4? The same way?
